llvm/include/llvm/Target/Target.td

```diff
@@ -619,6 +634,11 @@ class Instruction : InstructionEncoding {
   // Added complexity passed onto matching pattern.
   int AddedComplexity  = 0;
 
+  // Passed onto matching pattern and helps in skipping checks
+  // where mayLoad/mayStore is inferred in the instruction but doesn't
+  // mirror in the pattern.
+  bit ModelsInaccessibleMemThroughRegs = false;
```

Comment:
IIRC, this is needed because tablegen ISel patterns expect that if the pattern input has some kind of side effects (e.g. it reads inaccessible memory), then the output must also have side effects.

This is not true in our case, because some of our intrinsics read inaccessible mem (because they need to read state from control registers, which are invisible in the IR layer), but after ISel, those become instructions in which implicit reads of control registers are visible. So there are no more side effects.